### ISSUE TITLE
Implement @module

### ISSUE THREAD

**Issue by curvedmark** [curvedmark/roole#62]
```
@module .module with '--' {
    .class { margin: 0 }
}
```

->

```
.module--class {
    margin: 0;
}
```

<hr>

```
@module .button {
    display: inline-block;
    &.active {
        background: gray;   
        .icon {
            background: gray
        }
    }
}
```

->

```
.button {
    display: inline-block;
}
    .button.active {
        background: gray;
    }
        .button.active .button-icon {
            background: gray;
        }
```

This is an experimental feature, feel free to leave any ideas and suggestions.

**Comment by curvedmark** [comment]
Just realized a few drawbacks:
- When CSS frameworks are namespaced, their js code is likely to break, because the selector names are now changed, unless they also support selector namespacing in their js code, which is probably non-trivial.
- ~~Modules look a lot like nested rulesets, I'm not sure if it's confusing that with nested rulesets, selectors are joined with combinators, while with modules, selectors are never joined, only prefixed~~.
  
  Only the direct child rulesets' selectors are not joined, probably not that confusing.

**Comment by curvedmark** [comment]
Since `@module` no longer has to provide functionality like `&-foo` (#38), it can be greatly simplified:

```
@module name {
    // rules
}
```

Direct child properties are not allowed, all class selectors inside it will be prepend with `name-`, that's it.

**Comment by jamiebuilds** [comment]
And how will `&` selectors be handled within `@module`?

**Comment by curvedmark** [comment]
If `@module` is itself contained in a ruleset, then the `&` selector will use the parent ruleset's selector. In other words, `@module` is transparent.

Do you have any use case for using the module name as a class selector?

**Comment by curvedmark** [comment]
Landed in 5d56dbd14db72f0c2229265c5ac5689b326fc750
